episodes/06b-branches.md

```diff
@@ -0,0 +1,329 @@
+---
+title: Branches
+teaching: 20
+exercises: 0
+---
+
+::::::::::::::::::::::::::::::::::::::: objectives
+
+- Understand why branches are useful for:
+  - Working on separate tasks in the same repository concurrently
+  - Trying multiple solutions to a problem
+  - Check-pointing versions of code
+- Merge branches back into the main branch
+
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::: questions
+
+- What are branches?
+- How can I work in parallel using branches?
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+So far we've always been working in a straight timeline.
+However, there are times when we might want to keep
+our main work safe from experimental changes we are working on.
+To do this we can use branches to work on separate tasks in parallel
+without changing our current branch, `main`.
+
+We didn't see it before but the first branch made is called `main`.
+This is the default branch created when initializing a repository and
+is often considered to be the "clean" or "working" version of a
+repository's code.
+
+We can see what branches exist in a repository by typing
+
+```bash
+$ git branch
+```
+
+```output
+* main
+```
+
+
+The `*` indicates which branch we are currently on.
+
+In this lesson, Alfredo is trying to run an analysis
+and doesn't know if it will be faster in bash or python.
+To keep his main branch safe he will use separate branches
+for both bash and python analysis.
+Then he will merge the branch with the faster script
+into his main branch.
+
+First let's make the python branch.
+We use the same `git branch` command but now add the 
+name we want to give our new branch
+
+```bash
+$ git branch pythondev
+```
+
+We can now check our work with the `git branch` command.
+
+```bash
+$ git branch
+```
+
+```output
+* main
+  pythondev
+```
+
+We can see that we created the `pythondev` branch but we
+are still in the main branch.
+
+We can also see this in the output of the `git status` command.
+
+```bash
+$ git status
+```
+
+```output
+On branch main
+nothing to commit, working directory clean
+```
+
+To switch to our new branch we can use the `checkout` command
+we learned earlier and check our work with `git branch`.
+
+```bash
+$ git checkout pythondev
+$ git branch
+```
+
+```output
+  main
+* pythondev
+```
+
+Before we used the `checkout` command to checkout a file from a specific commit
+using commit hashes or `HEAD` and the filename (`git checkout HEAD <file>`). The
+`checkout` command can also be used to checkout an entire previous version of the
+repository, updating all files in the repository to match the state of a desired commit.
```

Comment:
Same as above: `checkout` is no longer taught in this lesson, so this is not relevant any more. The comparison can be done with `git restore` if needed.